metals/src/main/scala/scala/meta/internal/metals/ScaladocDefinitionProvider.scala

```diff
@@ -0,0 +1,339 @@
+package scala.meta.internal.metals
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ListBuffer
+import scala.util.Success
+import scala.util.Try
+
+import scala.meta.Defn
+import scala.meta.Pkg
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.inputs.Input
+import scala.meta.inputs.Position
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.tokens.Token.Comment
+
+import org.eclipse.lsp4j.TextDocumentPositionParams
+
+class ScaladocDefinitionProvider(
+    buffers: Buffers,
+    trees: Trees,
+    destinationProvider: DestinationProvider,
+) {
+
+  def definition(
+      path: AbsolutePath,
+      params: TextDocumentPositionParams,
+  ): Option[DefinitionResult] = {
+    for {
+      buffer <- buffers.get(path)
+      position <- params.getPosition().toMeta(Input.String(buffer))
+      symbol <- extractScalaDocLinkAtPos(buffer, position)
+      contextSymbols = getContext(path, position)
+      scalaMetaSymbols = symbol.toScalaMetaSymbols(contextSymbols)
+      _ = scribe.info(
+        s"looking for definition for scaladoc symbol: $symbol considering alternatives: ${scalaMetaSymbols
+            .map(_.showSymbol)
+            .mkString(", ")}"
+      )
+      definitionResult <- scalaMetaSymbols.collectFirst { sym =>
```

Comment:
I don't think so. According to scaladoc documentation if you have class and object it will prioritise the class. We do the same. Also we search: 1. this 2. this package 3. global, which is also compatible with how scopes work.